mubench.reviewsite/src/CSVHelper.php

```diff
@@ -0,0 +1,118 @@
+                "no_agreements", "total_agreements", "yes_no_agreements", "no_yes_agreements", "total_disagreements",
+                "kappa_p0", "kappa_pe", "kappa_score", "hits", "precision"];
+        } else {
+            $rows[] = ["detector", "project", "misuses", "potential_hits", "open_reviews", "need_clarification",
+                "yes_agreements", "no_agreements", "total_agreements", "yes_no_agreements", "no_yes_agreements",
+                "total_disagreements", "kappa_p0", "kappa_pe", "kappa_score", "hits", "recall"];
+        }
+        foreach ($stats as $key => $stat) {
+            $columns = [];
+            $columns[] = $stat->getDisplayName();
+            $columns[] = $stat->number_of_projects;
+            if (strcmp($experiment, "ex1") === 0) {
+                $columns[] = $stat->number_of_synthetics;
+            }
+            if (strcmp($experiment, "ex1") === 0 || strcmp($experiment, "ex3") === 0) {
+                $columns[] = $stat->number_of_misuses;
+            }
+            $kappa_p0 = round($stat->getKappaP0(), 3);
+            $kappe_pe = round($stat->getKappaPe(), 3);
+            $kappa_score = round($stat->getKappaScore(), 3);
+            if (strcmp($experiment, "ex2") === 0) {
+                $recall = round($stat->getPrecision() * 100, 1);
+            } else {
+                $recall = round($stat->getRecall() * 100, 1);
+            }
+
+            $columns[] = $stat->misuses_to_review;
+            $columns[] = $stat->open_reviews;
+            $columns[] = $stat->number_of_needs_clarification;
+            $columns[] = $stat->yes_agreements;
+            $columns[] = $stat->no_agreements;
+            $columns[] = $stat->getNumberOfAgreements();
+            $columns[] = $stat->yes_no_disagreements;
+            $columns[] = $stat->no_yes_disagreements;
+            $columns[] = $stat->getNumberOfDisagreements();
+            $columns[] = $kappa_p0;
+            $columns[] = $kappe_pe;
+            $columns[] = $kappa_score;
+            $columns[] = $stat->number_of_hits;
+            $columns[] = $recall;
+            $rows[] = $columns;
+        }
+        return $this->createCSV($rows);
+    }
+
+    private function createCSV($lines)
+    {
+        $imploded_lines = [];
+        foreach ($lines as $line) {
+            $imploded_lines[] = implode(',', $line);
+        }
+        return implode("\n", $imploded_lines);
+    }
+
+    public function createCSVFromRuns($runs)
+    {
+        $rows[] = ["sep=,"];
+        $rows[] = ["project", "version", "result", "number_of_findings", "runtime", "misuse", "decision",
+            "resolution_decision", "resolution_comment"];
+        $max_review_count = 2;
+        foreach ($runs as $run) {
+            $run_details =
+                [$run['project'], $run['version'], $run['result'], $run['number_of_findings'], $run['runtime']];
+            foreach ($run['misuses'] as $misuse) {
+                $columns = $run_details;
```

Comment:
Does this really copy the `$run_details` array? If it assigns by references this breaks when we have multiple misuses. As far as I see, all tests have but one misuse.